opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1621,20 +1621,44 @@ namespace Opm
         } else if (well.isInjector() ) {
             const Opm::Well2::InjectorCMode& current = well_state.currentInjectionControls()[well_index];
             const auto controls = well.injectionControls(summaryState);
+
+            Well2::InjectorType injectorType = controls.injector_type;
+            double scaling = 1.0;
+
+            const auto& pu = phaseUsage();
+
+            switch (injectorType) {
+            case Well2::InjectorType::WATER:
+            {
+                scaling = scalingFactor(pu.phase_pos[BlackoilPhases::Aqua]);
+                break;
+            }
+            case Well2::InjectorType::OIL:
+            {
+                scaling = scalingFactor(pu.phase_pos[BlackoilPhases::Liquid]);
+                break;
+            }
+            case Well2::InjectorType::GAS:
+            {
+                scaling = scalingFactor(pu.phase_pos[BlackoilPhases::Vapour]);
+                break;
+            }
+            default:
+                throw("Expected WATER, OIL or GAS as type for injectors " + well.name());
+            }
+
             switch(current) {
             case Well2::InjectorCMode::RATE:
             {
-                control_eq = getSegmentGTotal(0) * efficiencyFactor - controls.surface_rate;
+                control_eq = getSegmentGTotal(0) / scaling * efficiencyFactor - controls.surface_rate;
```

Comment:
I am not sure the usage of `efficiencyFactor` here is correct. But I did not review the previous PR related to this, there is something we can discuss here. 

And also, for the scaling factor here, it is probably better to use multiplication than division .

Reply:
> I am not sure the usage of efficiencyFactor here is correct.

Could you be more specific? What is wrong?

> And also, for the scaling factor here, it is probably better to use multiplication than division .

Do you mean. 

```
scaling = 1 / scalingFactor(pu.phase_pos[BlackoilPhases::Vapour]);

control_eq = getSegmentGTotal(0) * scaling * efficiencyFactor - controls.surface_rate;
```

Reply:
I agree about the scaling, multiply the control instead. That way the control equations are scaled similarly for all RATE targets.

Reply:
I will come back for the efficiency factor soon later. My current impression is that something is wrong here, but I can be wrong. I need to address something else first.

Reply:
OK. I have changed the scaling. If there is something wrong with the efficiency factor that will be another PR.

Reply:
The order of the scaling may matter. Now I new test case I got from Torbjørn does not run through due to singular matrix. The field case seems unaffected. I may have messed up something els, so I will double check. But we should not merge this before I confirm that the changes you suggested does not harm anything.

Reply:
> . Now I new test case I got from Torbjørn does not run through due to singular matrix. 

If the multiplication introduces singular matrix, there must be something else is wrong.

Reply:
You are right it is actually scaled up. I think we agree that the current scaling of the well equation (both the control and the flow) are not optimal. I still think this can be merged. We can change the placement of the scaling factor later when we find the real issue.